vm/src/stdlib/math.rs

```diff
@@ -657,6 +657,64 @@ fn math_remainder(x: IntoPyFloat, y: IntoPyFloat, vm: &VirtualMachine) -> PyResu
     Ok(x)
 }
 
+#[derive(FromArgs)]
+struct ProdArgs {
+    #[pyarg(positional)]
+    iterable: PyIterable<Either<PyFloatRef, PyIntRef>>,
+    #[pyarg(named, optional)]
+    start: OptionalArg<Either<PyFloatRef, PyIntRef>>,
+}
+fn math_prod(args: ProdArgs, vm: &VirtualMachine) -> PyResult<Either<BigInt, f64>> {
+    let iter = args.iterable;
+
+    let mut is_result_float = false;
+    let mut i_result: BigInt = BigInt::one();
+    let mut f_result: f64 = 1.0;
```

Comment:
we only need a single type of result, but for now, preparing both of them reagardless the type of input.